test/scripts/cloud-prepare/test.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env bash
+
+set -e -o pipefail
+
+source ${SCRIPTS_DIR}/lib/debug_functions
+source ${SCRIPTS_DIR}/lib/utils
+
+function verify_gw_topology() {
+    local expected_gateways="${1}"
+    local gateways_count
+
+    echo "Verifying ${expected_gateways} gateway nodes"
+    gateways_count=$(kubectl get nodes -l=submariner.io/gateway=true \
+        -o yaml | yq '.items | length' -)
+
+    if [[ "$expected_gateways" -ne "$gateways_count" ]]; then
+        echo "Expected ${expected_gateways} gateways nodes but detected ${gateways_count}"
+        return 1
+    fi
+
+    echo "Found expected number of gateways - ${expected_gateways}"
+}
+
+function remove_gw_labels() {
+    local gw_nodes
+
+    echo "Reset gateway nodes (unlabel)"
+    readarray -t gw_nodes < <(kubectl get nodes -l=submariner.io/gateway=true \
+        -o yaml | yq '.items[].metadata.name' -)
+
+    for node in "${gw_nodes[@]}"; do
+        kubectl label --overwrite nodes "$node" submariner.io/gateway-
+    done
+}
+
+echo "Prepare cluster"
+make clusters SETTINGS="$(dirname $0)/no_gw_defined.yml"
+declare_kubeconfig
+
+for scenario in no_gw_defined.yml two_gw_defined.yml; do
+    export SETTINGS="$(dirname $0)/${scenario}"
+
+    test_gateways=$(yq e '.gateways' $SETTINGS)
+    [[ "${test_gateways}" == null ]] && test_gateways=1
+
+    echo "Set gateway nodes according to ${scenario} scenario"
+    make cloud-prepare
+
+    load_settings
+    run_all_clusters verify_gw_topology "${test_gateways}"
+    run_all_clusters remove_gw_labels
```

Comment:
@mkolesnik 
Applied the suggestion.

Just added "load_settings" function to be able to connect to the clusters while checking the gateways.
And exporting the SETTINGS as environment variable.
Otherwise, the flow fails with the following error:
`ERROR: Expected env variable 'SETTINGS' can't be empty (Deployment settings file)`